target/bin/setup

```diff
@@ -0,0 +1,171 @@
+    arguments by appending ${WHITE}help${RESET} after your command. Currently, this does not work with all scripts.
+
+${RED}[${ORANGE}SUB${RED}]${ORANGE}COMMANDS${RESET}
+    ${LBLUE}COMMAND${RESET} email ${RED}:=${RESET}
+        ${0} email ${CYAN}add${RESET} <EMAIL ADDRESS> [<PASSWORD>]
+        ${0} email ${CYAN}update${RESET} <EMAIL ADDRESS> [<PASSWORD>]
+        ${0} email ${CYAN}del${RESET} [ OPTIONS${RED}...${RESET} ] <EMAIL ADDRESS> [ <EMAIL ADDRESS>${RED}...${RESET} ]
+        ${0} email ${CYAN}restrict${RESET} <add${RED}|${RESET}del${RED}|${RESET}list> <send${RED}|${RESET}receive> [<EMAIL ADDRESS>]
+        ${0} email ${CYAN}list${RESET}
+
+    ${LBLUE}COMMAND${RESET} alias ${RED}:=${RESET}
+        ${0} alias ${CYAN}add${RESET} <EMAIL ADDRESS> <RECIPIENT>
+        ${0} alias ${CYAN}del${RESET} <EMAIL ADDRESS> <RECIPIENT>
+        ${0} alias ${CYAN}list${RESET}
+
+    ${LBLUE}COMMAND${RESET} quota ${RED}:=${RESET}
+        ${0} quota ${CYAN}set${RESET} <EMAIL ADDRESS> [<QUOTA>]
+        ${0} quota ${CYAN}del${RESET} <EMAIL ADDRESS>
+
+    ${LBLUE}COMMAND${RESET} config ${RED}:=${RESET}
+        ${0} config ${CYAN}dkim${RESET} [ ARGUMENTS${RED}...${RESET} ]
+
+    ${LBLUE}COMMAND${RESET} relay ${RED}:=${RESET}
+        ${0} relay ${CYAN}add-domain${RESET} <DOMAIN> <HOST> [<PORT>]
+        ${0} relay ${CYAN}add-auth${RESET} <DOMAIN> <USERNAME> [<PASSWORD>]
+        ${0} relay ${CYAN}exclude-domain${RESET} <DOMAIN>
+
+    ${LBLUE}COMMAND${RESET} debug ${RED}:=${RESET}
+        ${0} debug ${CYAN}fetchmail${RESET}
+        ${0} debug ${CYAN}fail2ban${RESET} [unban <IP>]
+        ${0} debug ${CYAN}show-mail-logs${RESET}
+        ${0} debug ${CYAN}inspect${RESET}
+        ${0} debug ${CYAN}login${RESET} <COMMANDS>
+
+${ORANGE}EXAMPLES${RESET}
+    ${WHITE}./setup.sh email add test@domain.tld${RESET}
+        Add the email account ${WHITE}test@domain.tld${RESET}. You will be prompted
+        to input a password afterwards since no password was supplied.
+
+    ${WHITE}./setup.sh config dkim keysize 2048 domain 'whoami.com,whoareyou.org'${RESET}
+        Creates keys of length 2048 but in an LDAP setup where domains are not known to
+        Postfix by default, so you need to provide them yourself in a comma-separated list.
+
+    ${WHITE}./setup.sh config dkim help${RESET}
+        This will provide you with a detailed explanation on how to use the ${WHITE}
+        config dkim${RESET} command, showing what arguments can be passed and what they do.
+
+"
+}
+
+function _invalid_command
+{
+  echo "The command '${*}' is invalid.
+Use \`./setup.sh help\` to get an overview of all commands." >&2
+  exit 2
+}
+
+function _main
+{
+  case ${1:-} in
+
+    email )
+      case ${2:-} in
+        add      ) shift 2 ; addmailuser "${@:+$@}" ;;
+        update   ) shift 2 ; updatemailuser "${@:+$@}" ;;
+        del      ) shift 2 ; delmailuser "${@:+$@}" ;;
+        restrict ) shift 2 ; restrict-access "${@:+$@}" ;;
```

Comment:
```suggestion
        add      ) shift 2 ; addmailuser "${@}" ;;
        update   ) shift 2 ; updatemailuser "${@}" ;;
        del      ) shift 2 ; delmailuser "${@}" ;;
        restrict ) shift 2 ; restrict-access "${@}" ;;
```

Remove MacOS specific quirks, intruduced in #2001

Reply:
Ah, I did not think of this myself. Thanks!